pkg/webhook/pod/pod.go

```diff
@@ -80,18 +82,22 @@ func patchPodChanges(originalPod, currentPod *corev1.Pod) admission.Response {
 	currentNetworkAnnotation := currentPod.GetAnnotations()[pool_manager.NetworksAnnotation]
 	originalPodNetworkAnnotation := originalPod.GetAnnotations()[pool_manager.NetworksAnnotation]
 	if originalPodNetworkAnnotation != currentNetworkAnnotation {
-		annotationPatch := jsonpatch.NewPatch("replace", "/metadata/annotations", currentPod.GetAnnotations())
+		annotationPatch := jsonpatch.NewOperation("replace", "/metadata/annotations", currentPod.GetAnnotations())
 		kubemapcoolJsonPatches = append(kubemapcoolJsonPatches, annotationPatch)
 	}
 
 	log.Info("patchPodChanges", "kubemapcoolJsonPatches", kubemapcoolJsonPatches)
-	return admission.Response{
+	response := admission.Response{
 		Patches: kubemapcoolJsonPatches,
-		AdmissionResponse: admissionv1beta1.AdmissionResponse{
-			Allowed:   true,
-			PatchType: func() *admissionv1beta1.PatchType { pt := admissionv1beta1.PatchTypeJSONPatch; return &pt }(),
+		AdmissionResponse: admissionv1.AdmissionResponse{
+			Allowed: true,
 		},
 	}
+	if len(kubemapcoolJsonPatches) > 0 {
+		response.AdmissionResponse.PatchType = func() *admissionv1.PatchType { pt := admissionv1.PatchTypeJSONPatch; return &pt }()
```

Comment:
I think composite literal could be used here `&admissionv1.PatchTypeJSONPatch{}` which would look a bit better in my opinion, but not a must, of course